### DISCUSSION TITLE
[9일차 - 예진] 일급 객체

### DISCUSSION THREAD

**Discussion by ez1n** [zzangCS/zzangInterview#32] [프론트 면접 질문]
일급 객체란 무엇인가요?

**Comment by skyhanull** [comment]
일급객체의 일급은 다른 요소들과 아무런 차별이 없다라는 것을 의미합니다
위키백과에서는 "컴퓨터 프로그래밍 언어에서, 다른 객체들에 일반적으로 적용 가능한 연산을 모두 지원하는 객체."를 의미합니다.
일급 객체는 3가지의 조건을 충족해야 하는데
1. 모든 일급 객체는 변수나 데이터에 담을 수 있어야 한다.
2. 모든 일급 객체는 함수의 파라미터로 전달 할 수 있어야 한다.
3. 모든 일급 객체는 함수의 리턴값으로 사용 할수 있어야 한다.

**Comment by ez1n** [comment]
- **일급객체**
 변수나 데이터에 담을 수 있어야 하고,
 함수의 파라미터로 전달할 수 있어야 하며
 함수의 리턴 값으로 사용할 수 있는 객체

세 가지 조건을 모두 충족한 객체를 일급 객체라고 한다.
